libp2p/identity/identify/protocol.py

```diff
@@ -0,0 +1,45 @@
+import logging
+from typing import Sequence
+
+from multiaddr import Multiaddr
+
+from libp2p.crypto.keys import PublicKey
+from libp2p.network.stream.net_stream_interface import INetStream
+from libp2p.typing import StreamHandlerFn, TProtocol
+
+from .pb.identify_pb2 import Identify
+
+ID = "/ipfs/id/1.0.0"
+PROTOCOL_VERSION = "ipfs/0.1.0"
+# TODO dynamically generate the agent version
+AGENT_VERSION = "py-libp2p/alpha"
+logger = logging.getLogger("libp2p.identity.identify")
+
+
+def _multiaddr_to_bytes(maddr: Multiaddr) -> bytes:
+    return maddr.to_bytes()
+
+
+def identify_handler_for(
+    public_key: PublicKey, laddrs: Sequence[Multiaddr], protocols: Sequence[TProtocol]
+) -> StreamHandlerFn:
+    async def handle_identify(stream: INetStream) -> None:
+        peer_id = stream.muxed_conn.peer_id
+        logger.debug("received a request for % from %", ID, peer_id)
```

Comment:
I got this error: `ValueError: unsupported format character '%'`

Reply:
Those need to be `%s`, not just plain `%`

Reply:
good catch!